File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        """
            *
            7,1,5,3,6,4
        
        """
        
        lowest = prices[0]
        
        profit = 0
        
        for price in prices:
            
            if price < lowest:
                lowest = price
                
            profit = max(profit,price-lowest)
            
        
        return profit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        
        int ptr1 =0;
        int ptr2 =0;
        
        int profit =0;
        
        while(ptr2 < prices.length){
            
            if(profit < (prices[ptr2] - prices[ptr1])){
                profit = prices[ptr2] - prices[ptr1];
            }
            
            if(prices[ptr2] < prices[ptr1]){
                ptr1=ptr2;
            }
            

            
            ptr2 += 1;
        }
        
        return profit;
    }
}